### ISSUE TITLE
Author only API key not displaying all tags after server update to 3.2.1

### ISSUE THREAD

**Issue by LINCCLIBS** [shlinkio/shlink#1537]
#### How Shlink is set-up

* Shlink Version: 3.2.1
* PHP Version: 8.1.9
* How do you serve Shlink: Self-hosted nginx
* Database engine used: MariaDB

#### Summary

After updating to Shlink verison 3.2.1, an API "author only" API key we use only shows 1 tag, when looking at the "Manage tags" area of app.shlink.io. 

#### Current behavior

There is only one tag displaying in app.shlink.io under "Manage tags", where there should be 20+
![image](https://user-images.githubusercontent.com/9955397/188945617-5f58a8f9-1197-4dc0-9050-89398975b734.png)

If you look at the "List short URLs", you can see the other tags listed in the "Tags" column
![image](https://user-images.githubusercontent.com/9955397/188946104-3f94d10c-afa4-496f-bb67-d97448ba6ceb.png)

#### Expected behavior

All tags should display and be manageable from the "Manage tags" area of app.shlink.io

#### How to reproduce

This happened when I created an author only api key, created several tags and short links, then updated the server to version 3.2.1.

Thank you for looking into this, and please let me know if I can help in any way.

**Comment by acelaya** [comment]
> then updated the server to version 3.2.1

What version were you using before?

**Comment by LINCCLIBS** [comment]
It had been quite a while (probably too long) since I updated the server.  I believe the previous version was 2.6.1.

**Comment by acelaya** [comment]
I'm struggling to reproduce this. For me it works, and I'm not sure what I'm doing differently,

Also, Shlink has several E2E tests covering API key roles, including the different endpoints to list tags, and they have bee passing since implemented.

Can you share the output of running `bin/cli api-key:list` on your server? Make sure to blur the actual API keys values.

**Comment by LINCCLIBS** [comment]
Yes, below is the output of bin/cli api-key:list.    I am checking with some of the other people who use author only keys to see if anyone other than the one I originally reported has this issue.  I took a look at a couple others that use tags and couldn't immediately detect a problem.  The one that I originally reported is the seventh from the bottom on this list.  It does seem strange that one key would act differently than the others.  Thanks again for your time in looking into this.  

![shlink-api-keys-blurred](https://user-images.githubusercontent.com/9955397/189209732-0317d796-df6e-4a81-b086-c0ae1c070a5f.png)

**Comment by acelaya** [comment]
And if you create a new author API key, does this one work as expected?

**Comment by LINCCLIBS** [comment]
Yes, I just created an author API key, created a couple of short urls, and assigned tags, and everything seems to be working as expected.

**Comment by acelaya** [comment]
Ok, in that case I will try to setup Shlink 2.6, create API keys, short URLs and tags, and then update to v3.2

**Comment by acelaya** [comment]
I have set-up a Shlink 2.6.2 with MySQL, created an admin API key and an author API key, created several short URLs and tags with both of them. Then I have updated to Shlink 3.2.1, and everything continued working as expected.

I have tried a couple of times with different combinations, and everything was working as expected.

I'm afraid I will have to close this for now as not reproducible.

If you manage to create a failing test case (inside `module/Rest/test-api` there are several tests covering different API keys and permissions) I will continue investigating.

**Comment by cobenash** [comment]
I faced the same issue. All the tags of the authors shows 8 tags but actually more.

**Comment by LINCCLIBS** [comment]
> I faced the same issue. All the tags of the authors shows 8 tags but actually more.

I haven't found a way to fix this at this point.  The end user who had this issue on our server is now using an admin key, which can see everything.

**Comment by acelaya** [comment]
Reopening in order to try to investigate a bit further. However, any help reproducing it will be a massive help.

**Comment by acelaya** [comment]
I have recently made some changes to the query which resolves the list of tags. It could help here.

However I did it blindly, as I still haven't been able to reproduce the problem.

Asking again for help, if someone would be willing to provide a database dump that could help me get to the bottom of this.

**Comment by LINCCLIBS** [comment]
I would be happy to help with this.  We are still running release 3.2.1. Would it be better to update to the latest version first?  If not, I can do an sql dump now.  What would be the best way to get it to you?  Thanks!

**Comment by acelaya** [comment]
Awesome!

Please, generate the dump before updating Shlink, and send it to my email.

My email is my name and surname all together, at gmail.com

So if my name was John Doe, it would be johndoe@gmail.com

I hope that's clear enough 😅

**Comment by LINCCLIBS** [comment]
OK, I sent you my database.  Hopefully that will help!
Thanks,
David

**Comment by acelaya** [comment]
Received!
Let's see what I can find out.

**Comment by acelaya** [comment]
Ok, I have very good news to share. Thanks to @LINCCLIBS's help I was able to reproduce the issue. It's reproducible on any stable Shlink version (including 3.5.2).

However, the issue is not reproducible in `latest`, which means the changes I mentioned [here](https://github.com/shlinkio/shlink/issues/1537#issuecomment-1478988591) have indeed fixed this :)

I will add this to v3.5.3 CHANGELOG, and close the issue as fixed.

I will release v3.5.3 later this week or at the beginning of the next one.

**Comment by LINCCLIBS** [comment]
I can confirm that this update solved the problem, thank you again @acelaya for this fix and for all your work on Shlink!
